### PR TITLE
SceneOptimizer: Introduce `optimizer.toBatchedMesh()`

### PR DESCRIPTION
**Pull request by RenaudRohlinger**
**Description**
Adds a new utility helper that automatically optimizes a scene by converting compatible meshes into BatchedMeshes.

Example usage:
```js
import { optimizeSceneToBatchedMesh } from 'three/addons/utils/BatchedMeshUtils.js';

loader.load( 'dungeon_warkarma.glb', async function ( gltf ) {

	model = gltf.scene;

        // optimizes the scene by converting compatible meshes into BatchedMeshes.
	const optimizer = new SceneOptimizer( model, { debug: true } );
	optimizer.toBatchedMesh();

	scene.add( model );

} );
```

image

*This contribution is funded by [Utsubo](https://utsubo.com)*

### CODE REVIEW

File: examples/jsm/utils/BatchedMeshUtils.js [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+
+			const map = material[ prop ];
+			if ( ! map ) return 0;
+			return `${map.uuid}_${map.offset.x}_${map.offset.y}_${map.repeat.x}_${map.repeat.y}_${map.rotation}`;
+
+		} )
+		.join( '|' );
+
+	// Build physical properties hash
+	const physicalProps = [
```
Comment (Mugen87) [inline examples/jsm/utils/BatchedMeshUtils.js:69]: What if the scene contains instances of `*NodeMaterial` classes (e.g. `MeshStandardNodeMaterial`). 

It looks like properties like `normalNode` or `roughnessNode` are not honored yet.

**Comment by Mugen87** [comment]
I think it's a good idea to implement such a utility since it helps us exploring the topic of "auto-batching" and "auto-instancing".

File: examples/jsm/utils/BatchedMeshUtils.js (hunk 1 above)
Comment (RenaudRohlinger) [reply examples/jsm/utils/BatchedMeshUtils.js:69]: I originally envisioned this utility as a preprocessing step, primarily for 3D assets like GLB files. For MeshStandardNodeMaterial, the material would ideally be applied afterward or the BatchedMesh handled manually, without relying on this utility.

Attempting to integrate the Node system here could lead to maintenance issues and may not be compatible with the WebGLRenderer.

**Comment by sunag** [comment]
Wouldn't it be better to turn this into a class so we can easily extend it when we add instancing support?

```js
const optimizer = new SceneOptimizer( scene );
optimizer.toBatchedMesh();
// optimizer.toInstancingMesh(); // soon
````

**Comment by RenaudRohlinger** [comment]
@sunag Love the idea! Refactored my work accordingly to your comment.

**Pull request merged**